src/modules/group/group.repository.ts

```diff
@@ -1,15 +1,74 @@
-import { eq } from "drizzle-orm";
-
+import { AuthRoles } from "@/common/enums";
 import { db } from "@/db/adapter";
 import type { TInsertGroupSchema } from "@/db/schemas/group.model";
 import groupModel from "@/db/schemas/group.model";
+import type { TSelectUserSchema } from "@/db/schemas/user.model";
+import type { SQL } from "drizzle-orm";
+import { and, asc, desc, eq, ilike, sql } from "drizzle-orm";
+
+import type { TGroupQuerySchema } from "./group.schema";
 
 export async function createGroupRepository(groupPayload: TInsertGroupSchema) {
   const [group] = await db.insert(groupModel).values(groupPayload).returning();
 
   return group;
 }
 
+export async function getAllGroupsRepository(queryParams: TGroupQuerySchema, userDetails: TSelectUserSchema) {
+  const { page, limit, status, sortOrder, search } = queryParams;
+  const offset = (page - 1) * limit;
+  const whereConditions: SQL<unknown>[] = [];
+
+  if (userDetails.role === AuthRoles.USER) {
+    whereConditions.push(eq(groupModel.creatorId, userDetails.id));
+  }
+
+  if (status !== undefined) {
+    whereConditions.push(eq(groupModel.status, status));
+  }
+
+  if (search) {
+    whereConditions.push(ilike(groupModel.name, `%${search.toLowerCase()}%`));
```

Comment:
Search has to be done using PostgreSQL's full text search, for reference please check category search.